components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt

```diff
@@ -73,6 +73,10 @@ class WebExtensionBrowserMenu internal constructor(
             val extensions = state.extensions.values.toList()
             extensions.filter { it.enabled }.sortedBy { it.name }
                 .forEach { extension ->
+                    if (!extension.allowedInPrivateBrowsing && tab?.content?.private == true) {
+                        return@forEach
```

Comment:
You don't want to return here / just continue with the next extension. Another extension could be allowed in private browsing and should be applied.